### ISSUE TITLE
ros name remappings could allow remapping an entire namespace

### ISSUE THREAD

**Issue by jbohren** [ros/ros_comm#209]
There are a lot of meta-protocols implemented on top of ROS topics (image transport / actionlib / etc) in which several topics are encapsulated in a namespace. Currently, ROS tools can easily remap topics, but they can only remap a topic if the specification goes all the way to the leaf of the namespace tree. 

For example, if a user wanted ro remap `/my/foo/camera/image_raw` and `/my/foo/camera/camera_info` to `/your/foo/camera/image_raw` and `/my/foo/camera/camera_info`, respectively, then both of these remappings would need to be specified, instead of being able to just specify `/my/foo/camera:=/your/foo/camera`.

It would be really useful to add this capability to the ROS remapping specification.

**Comment by forrestv** [comment]
I agree. It was very surprising that it didn't already work like this.

**Comment by jbohren** [comment]
@forrestv Indeed!

**Comment by tlind** [comment]
I agree this would be extremely helpful.

**Comment by corot** [comment]
:+1:   Helpful and elegant!

**Comment by valschmidt** [comment]
I also think it would be a grand idea.

**Comment by corot** [comment]
It is possible to remap full actions (5 topics)... I suppose it would be an extension of that functionallity

**Comment by corot** [comment]
There are no plans to implement this feature for future version?

**Comment by dirk-thomas** [comment]
@corot As the milestone "untargeted" indicates the maintainers will likely not have time to implement this new feature. Please consider providing a pull request if it is of value for your use case.

**Comment by luator** [comment]
I may be able to work on this (depends on how much effort it is). Can you tell me where in the code the remapping is actually happening?

**Comment by dirk-thomas** [comment]
Remappings are implemented in the [node handle class](https://github.com/ros/ros_comm/blob/15fcd0c754ca1dbd5a38c208b61dd3e261fa585d/clients/roscpp/src/libros/node_handle.cpp).

**Comment by sloretz** [comment]
ROS 1 is end-of-life (EOL) as of today, May 31st 2025. I am archiving this repository because:

* it only supports ROS 1
* it isn't needed anymore in ROS 2

[If you still rely on ROS 1, read this page to learn about your options.](https://bit.ly/NoeticEOL)
